src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/BuildOutput.cs

```diff
@@ -49,8 +49,11 @@ public bool IsTargetSkipped (string target)
 			if (!Builder.LastBuildOutput.Contains (target))
 				throw new ArgumentException (string.Format ("Target '{0}' is not even in the build output.", target));
 			return Builder.LastBuildOutput.Contains (string.Format ("Target {0} skipped due to ", target))
-				      || Builder.LastBuildOutput.Contains (string.Format ("Skipping target \"{0}\" because its outputs are up-to-date", target))
-				      || Builder.LastBuildOutput.Contains ($"Skipping target \"{target}\" because all output files are up-to-date");
+					|| Builder.LastBuildOutput.Contains (string.Format ("Skipping target \"{0}\" because it has no outputs.", target))
```

Comment:
I don't think this needs to be addressed in this PR, but I have to wonder...

`Builder.LastBuildOutput` is a `string` which contains the `msbuild /v:diag` build output, which could be many *megabytes* in size, right? And this is doing possibly 5 linear searches against that output.

How does that perform?

Would...regular expressions be better?

```csharp
var toMatch = new[]{
    $"Skipping target \"{target}\" because it has no outputs.",
    $"Target \"{target}\" skipped, due to",
    $"Skipping target \"{target}\" because its outputs are up-to-date",
    $"target {target}, skipping",
    $"Skipping target \"{target}\" because all output files are up-to-date",
};
var r = new Regex (string.Join ("|", toMatch));
var m = r.Match (Builder.LastBuildOutput);
return m.Success;
```